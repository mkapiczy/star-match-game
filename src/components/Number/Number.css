.number {
    margin-right: 5px;
    background-color: #aeaeae;
    border: none;
    color: white;
    text-align: center;
    text-decoration: none;
    border-radius: 50%;
    width: 5vmin;
    height: 5vmin;
}

.number.not-available {
    background-color: #3F51B5;
}

.number:not(.not-available):hover {
    background-color: #d7d7d7;
}