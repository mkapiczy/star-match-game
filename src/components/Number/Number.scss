.number {
  margin-right: 0.25em;
  margin-left: 0.25em;
  background-color: #aeaeae;
  border: none;
  color: white;
  text-align: center;
  text-decoration: none;
  border-radius: 50%;
  width: 5vmin;
  height: 5vmin;
  font-size: 2.5vmin;

  &.not-available {
    background-color: #3F51B5;
  }

  &:not(.not-available):hover {
    background-color: #d8d8d8;
  }
}