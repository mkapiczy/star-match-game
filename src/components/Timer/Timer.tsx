import React from 'react';

interface Props {
    timer: number
}

const Timer: React.FC<Props> = (props: Props) => {
    return (
        <div className="timer" style={{fontWeight: "bold"}}>Time left: {props.timer}</div>
    );
}

export default Timer;