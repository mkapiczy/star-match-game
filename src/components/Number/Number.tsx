import React from 'react';
import './Number.scss';
import classNames from "classnames";

interface Props {
    numberValue: number,
    onNumberClick: (numberValue: number) => void;
    isAvailable: boolean
}

const Number: React.FC<Props> = (props: Props) => {
    return (
        <button key={props.numberValue} className={classNames('number', {'not-available': !props.isAvailable})}
                disabled={!props.isAvailable} onClick={() => props.onNumberClick(props.numberValue)}>
            {props.numberValue}
        </button>
    )
}

export default Number;