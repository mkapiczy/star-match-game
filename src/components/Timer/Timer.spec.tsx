import React from 'react';
import Timer from "./Timer";
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import Number from "../Number/Number";

describe("Timer component snapshot tests", () => {
    it("should match snapshot", () => {
        const component = renderer.create(
            <Timer timer={5}/>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})

describe('Timer component tests', () => {
    it("should render correct timer value", () => {
        const timerValue = 9
        const wrapper = mount( <Timer timer={timerValue}/>)
        const timer = wrapper.find('.timer')
        expect(timer.text()).toBe(`Time left: ${timerValue}`)

    })
});