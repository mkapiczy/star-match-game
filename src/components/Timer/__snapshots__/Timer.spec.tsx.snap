// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer component snapshot tests should match snapshot 1`] = `
<div
  className="timer"
  style={
    Object {
      "fontWeight": "bold",
    }
  }
>
  Time left: 
  5
</div>
`;
